### PR TITLE
Maintenance refactoring

### PR DESCRIPTION
**Pull request by xprazak2**
Extracts rule results and test result
into a separate concenrns.

Improves how newlines are removed from text.

### CODE REVIEW

File: lib/openscap_parser/test_result.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module OpenscapParser
```
Comment (akofink) [inline lib/openscap_parser/test_result.rb]:
```suggestion
# frozen_string_literal: true

module OpenscapParser
```

File: lib/openscap_parser/rule.rb [hunk 2]
```diff
@@ -20,11 +20,13 @@ def title
...
     def description
-      @description ||= @rule_xml.at_css('description').text.delete("\n")
+      rule_node ||= @rule_xml.at_css('description')
+      @description ||= newline_to_whitespace(rule_node.text) if rule_node && rule_node.text
     end
 
     def rationale
-      @rationale ||= @rule_xml.at_css('rationale').children.text.delete("\n")
+      rationale_node ||= @rule_xml.at_css('rationale')
+      @rationale ||= newline_to_whitespace(rationale_node.children.text) if rationale_node && rationale_node.children && rationale_node.children.text
```
Comment (akofink) [inline lib/openscap_parser/rule.rb]: This line is too long. I guess we still need to turn on rubocop in this repo :see_no_evil:

**Comment by xprazak2** [comment]
Updated with suggested changes.

File: lib/openscap_parser/rule.rb [hunk 3]
```diff
@@ -35,10 +39,15 @@ def references
 
     def identifier
       @identifier ||= {
-        label: @rule_xml.at_css('ident')&.text,
+        label: @rule_xml.at_css('ident') && @rule_xml.at_css('ident').text,
```
Comment (akofink) [inline lib/openscap_parser/rule.rb]: Why this change? You don't like the new `&.` syntax?

Comment (dLobatog) [reply lib/openscap_parser/rule.rb]: Perhaps the proxy has to run on an older version of ruby...?

Comment (akofink) [reply lib/openscap_parser/rule.rb]: Yeah - I'm not blocking this PR on it, just curious.

Comment (akofink) [reply lib/openscap_parser/rule.rb]: Yeah - I'm not blocking this PR on it, just curious.

**Comment by akofink** [comment]
Closing and reopening to trigger tests

File: lib/openscap_parser/rule.rb (hunk 3 above)
Comment (xprazak2) [reply lib/openscap_parser/rule.rb]: Exactly, proxies may have older versions of ruby.
